test/functional/api/vim_spec.lua

```diff
@@ -178,6 +178,14 @@ describe('API', function()
       -- Verify NO hit-enter prompt.
       eq({mode='n', blocking=false}, nvim("get_mode"))
     end)
+
+    it('Does not cause heap buffer overflow with large output', function()
+      if not os.getenv('ASAN_OPTIONS') then
+        pending('should run with ASAN build', function() end)
+      end
```

Comment:
why not always run it?

Reply:
Thought it would unnecessarily slow down non-ASAN builds, but yeah this is premature optimization (if any) :)

Reply:
Well, it takes quite some time (813ms).
Not much by itself, but this all adds up.

We could maybe have a tag for those kind of slower regression tests?

Reply:
This PR for now is fine for me..!